File: app/src/main/java/com/my/televip/features/other/RemovesContentSaving.java
package com.my.televip.features.other;


import com.my.televip.Class.ClassNames;
import com.my.televip.Clients.ClientManager;
import com.my.televip.Configs.ConfigManager;
import com.my.televip.base.BaseMethodHook;
import com.my.televip.hooks.HMethod;
import com.my.televip.Class.ClassLoad;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;
import com.my.televip.logging.Logger;

public class RemovesContentSaving {

    public static boolean isEnable = false;

    public static void init() {
        try {
            if (!isEnable) {
                isEnable = true;

                if (ClassLoad.getClass(ClassNames.MESSAGES_CONTROLLER) != null) {
                    HMethod.hookMethod(ClassLoad.getClass(ClassNames.MESSAGES_CONTROLLER), Obfuscate.getMethodName("MessagesController", "isChatNoForwards"), ArgsResolver.merge("isChatNoForwards", new Class[]{ClassLoad.getClass(ClassNames.TLRPC_CHAT)}, new BaseMethodHook() {
                        @Override
                        protected void beforeMethod(MethodHookParam param) {
                            if (ConfigManager.removesContentSaving.isEnable())
                                param.setResult(false);
                        }
                    }));
                }

                if (ClassLoad.getClass(ClassNames.CHAT_ACTIVITY) != null && !ClientManager.is(ClientManager.Client.NagramXF)) {
                    HMethod.hookMethod(ClassLoad.getClass(ClassNames.CHAT_ACTIVITY), Obfuscate.getMethodName("ChatActivity", "hasSelectedNoforwardsMessage"), new BaseMethodHook() {
                        @Override
                        protected void beforeMethod(MethodHookParam param) {
                            if (ConfigManager.removesContentSaving.isEnable())
                                param.setResult(false);
                        }
                    });
                }

                if (ClassLoad.getClass(ClassNames.MESSAGE_OBJECT) != null) {
                    HMethod.hookMethod(ClassLoad.getClass(ClassNames.MESSAGE_OBJECT), Obfuscate.getMethodName("MessageObject", "canForwardMessage"), new BaseMethodHook() {
                        @Override
                        protected void beforeMethod(MethodHookParam param) {
                            if (ConfigManager.removesContentSaving.isEnable())
                                param.setResult(true);
                        }
                    });
                }
            }
        } catch (Throwable t) {
            Logger.e(t);
        }
    }

}

File: app/src/main/java/com/my/televip/features/other/RemovesContentSaving.kt
package com.my.televip.features.other

import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.clients.ClientManager
import com.my.televip.configs.ConfigManager
import com.my.televip.base.BaseMethodHook
import com.my.televip.hooks.HMethod
import com.my.televip.logging.Logger
import com.my.televip.obfuscate.ArgsResolver
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

object RemovesContentSaving {

    @JvmField
    var isEnable: Boolean = false

    @JvmStatic
    fun init() {
        try {
            if (!isEnable) {
                isEnable = true

                val mcClass = ClassLoad.getClass(ClassNames.MESSAGES_CONTROLLER)
                if (mcClass != null) {
                    val types: Array<Class<*>> = arrayOf(ClassLoad.getClass(ClassNames.TLRPC_CHAT)!!)
                    val merged = ArgsResolver.merge("isChatNoForwards", types, object : BaseMethodHook() {
                        override fun beforeMethod(param: MethodHookParam) {
                            if (ConfigManager.removesContentSaving?.isEnable == true) {
                                param.result = false
                            }
                        }
                    })
                    if (merged != null) {
                        XposedHelpers.findAndHookMethod(
                            mcClass,
                            Obfuscate.getMethodName("MessagesController", "isChatNoForwards"),
                            *merged
                        )
                    }
                }

                val chatClass = ClassLoad.getClass(ClassNames.CHAT_ACTIVITY)
                if (chatClass != null && !ClientManager.`is`(ClientManager.Client.NagramXF)) {
                    HMethod.hookMethod(
                        chatClass,
                        Obfuscate.getMethodName("ChatActivity", "hasSelectedNoforwardsMessage"),
                        object : BaseMethodHook() {
                            override fun beforeMethod(param: MethodHookParam) {
                                if (ConfigManager.removesContentSaving?.isEnable == true) {
                                    param.result = false
                                }
                            }
                        }
                    )
                }

                val msgClass = ClassLoad.getClass(ClassNames.MESSAGE_OBJECT)
                if (msgClass != null) {
                    HMethod.hookMethod(
                        msgClass,
                        Obfuscate.getMethodName("MessageObject", "canForwardMessage"),
                        object : BaseMethodHook() {
                            override fun beforeMethod(param: MethodHookParam) {
                                if (ConfigManager.removesContentSaving?.isEnable == true) {
                                    param.result = true
                                }
                            }
                        }
                    )
                }
            }
        } catch (t: Throwable) {
            Logger.e(t)
        }
    }
}
